File: payment_program.py
import random

def generate_payment_slips_python():
    """
    Generates payment slips for workers, assigns employee levels, and handles potential errors.
    """
    num_workers = 400
    workers = []

    # Dynamically create a list of workers
    for i in range(1, num_workers + 1):
        worker_id = f"HRCC_W{i:04d}"
        salary = round(random.uniform(5000, 35000), 2) # Salaries between $5,000 and $35,000
        gender = random.choice(["Male", "Female"])
        workers.append({"id": worker_id, "salary": salary, "gender": gender, "level": "N/A"})

    print("Generating Payment Slips (Python):\n")

    for worker in workers:
        try:
            employee_level = ""
            if 10000 < worker["salary"] < 20000:
                employee_level = "A1"
            elif 7500 < worker["salary"] < 30000 and worker["gender"] == "Female":
                employee_level = "A5-F"
            else:
                employee_level = "B2" # Default level for others

            worker["level"] = employee_level

            print(f"Worker ID: {worker['id']}")
            print(f"Salary: ${worker['salary']:.2f}")
            print(f"Gender: {worker['gender']}")
            print(f"Employee Level: {worker['level']}")
            print("-" * 30)

        except Exception as e:
            print(f"Error processing worker {worker['id']}: {e}")
            print("-" * 30)

if __name__ == "__main__":
    generate_payment_slips_python()

File: payment_program.R
# Function to generate payment slips in R
generate_payment_slips_r <- function() {
  num_workers <- 400
  workers <- data.frame(
    id = character(num_workers),
    salary = numeric(num_workers),
    gender = character(num_workers),
    level = character(num_workers),
    stringsAsFactors = FALSE
  )

  # Dynamically create a list of workers
  for (i in 1:num_workers) {
    worker_id <- sprintf("HRCC_W%04d", i)
    salary <- round(runif(1, 5000, 35000), 2) # Salaries between $5,000 and $35,000
    gender <- sample(c("Male", "Female"), 1)
    
    workers$id[i] <- worker_id
    workers$salary[i] <- salary
    workers$gender[i] <- gender
    workers$level[i] <- "N/A" # Initialize level
  }

  print("Generating Payment Slips (R):\n")

  for (i in 1:nrow(workers)) {
    tryCatch({
      employee_level <- ""
      if (workers$salary[i] > 10000 && workers$salary[i] < 20000) {
        employee_level <- "A1"
      } else if (workers$salary[i] > 7500 && workers$salary[i] < 30000 && workers$gender[i] == "Female") {
        employee_level <- "A5-F"
      } else {
        employee_level <- "B2" # Default level for others
      }
      
      workers$level[i] <- employee_level
      
      cat(sprintf("Worker ID: %s\n", workers$id[i]))
      cat(sprintf("Salary: $%.2f\n", workers$salary[i]))
      cat(sprintf("Gender: %s\n", workers$gender[i]))
      cat(sprintf("Employee Level: %s\n", workers$level[i]))
      cat(sprintf("%s\n", paste(rep("-", 30), collapse = "")))
      
    }, error = function(e) {
      cat(sprintf("Error processing worker %s: %s\n", workers$id[i], e$message))
      cat(sprintf("%s\n", paste(rep("-", 30), collapse = "")))
    })
  }
}

# Run the R function
generate_payment_slips_r()